common/logging/logging.go

```diff
@@ -0,0 +1,49 @@
+package logging
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/peer"
+)
+
+func LoggingInterceptor() grpc.ServerOption {
+	return grpc.UnaryInterceptor(logInterceptor)
+}
+
+func loggerFromContext(ctx context.Context) *logrus.Entry {
+	// TODO: anonymize the addresses. cryptopan?
+	if peerInfo, ok := peer.FromContext(ctx); ok {
+		return log.WithFields(logrus.Fields{"peer_addr": peerInfo.Addr})
+	}
+	return log.WithFields(logrus.Fields{"peer_addr": "unknown"})
+}
+
+func logInterceptor(
+	ctx context.Context,
+	req interface{},
+	info *grpc.UnaryServerInfo,
+	handler grpc.UnaryHandler,
+) (interface{}, error) {
+	reqLog := loggerFromContext(ctx)
+	start := time.Now()
+
+	resp, err := handler(ctx, req)
+
+	entry := reqLog.WithFields(logrus.Fields{
+		"method":   info.FullMethod,
+		"duration": time.Since(start),
+		"error":    err,
+	})
```

Comment:
It doesn't need to be addressed in this PR, but I opened a ticket to disable this verbose logging of when RPCs get called: https://github.com/zcash/lightwalletd/issues/180

Reply:
Yeah, we should resolve that in another issue.